lib/TeleIrc.js

```diff
@@ -440,13 +447,38 @@ class TeleIrc {
 
   /**
    * Sends a message to the IRC channel.
-   * @param {String} message - The message to send to the IRC channel.
+   * @param {(String|Object)} input - The message to send to the IRC channel.
+   * @param {String} input.from - Name of the user from which the message was received.
+   * @param {String} input.message - Actual message to be sent.
+   * 
+   * If `input` is provided as string, message is forwarded as-is. Otherwise, a prefix is
```

Comment:
I'm not sure this is the place for this logic. It would fit better in https://github.com/RITlug/teleirc/blob/master/lib/TelegramHandlers/TgMessageHandler.js#L33, which already has all the data handy for assembling a message. The message splitting logic you're adding looks like it can pretty much just be moved to the TgMessageHandler, swapping out the `ircbot.say` call for `this._action`. Moving it would also save a little work since you then don't need to worry about other data types.

As-is, this looks like a separation of concerns problem. There are two reasons this caught my attention:
- This is a method that originally took a prebuilt piece of data and passed it along to somewhere else. Here, it gained the need to know more things and is now doing work that was originally done elsewhere. Design-wise for teleirc, message formatting should generally go in the TelegramHandlers (which would format to send to IRC) or in the IrcHandlers (which would format to send to Telegram).
- A method that gains a need to handle an additional data type in the same parameter makes me raise an eyebrow. There can be valid reasons for this to happen (extensions to a legacy API, for example) but it's often a sign of a larger design problem.

Reply:
I don't really agree:

> Design-wise for teleirc, message formatting should generally go in the TelegramHandlers (which would format to send to IRC)

The fact, that messages have certain length limitations is the issue on the IRC side. Why should Telegram part read IRC configuration options? What if you'd like to switch IRC backend to a different one, which doesn't have that limitation or has a different value?

> A method that gains a need to handle an additional data type in the same parameter makes me raise an eyebrow.

True - I left this as is to keep the change small and easily reviewable, for future refactoring.

But, in the end, it's your codebase, I will do the requested changes and you'll decide if you prefer my old approach or not :)

Reply:
Heh, the problem with dynamic languages.

I wouldn't think of it as passing in an IRC Config object, but passing in an object that implements an interface that contains the prefix, suffix, and maximum length; an interface that IRC Config just so happens in implement.  A theoretical SlackConfig object or a WhatsAppConfig object could implement the same "interface" and be passed in instead.  Its why the parameter in the constructor was originally called "prefixSuffixConfig" instead of "ircConfig".